src/http_server/api/v1/health.c

```diff
@@ -245,6 +303,104 @@ static int cleanup_metrics()
     return c;
 }
 
+static int check_throughput_health(uint64_t in_records,
+                                   uint64_t out_records,
+                                   struct mk_list *sample_list,
+                                   int sample_count,
+                                   double out_in_ratio_threshold) {
+    struct flb_time tp;
+    uint64_t timestamp_seconds;
+    uint64_t in_rate;
+    uint64_t out_rate;
+    struct mk_list *tmp;
+    struct mk_list *head;
+    double out_in_ratio;
+    struct flb_hs_throughput_sample *entry;
+    struct flb_hs_throughput_sample *prev;
+    struct flb_hs_throughput_sample *sample;
+    struct flb_hs_throughput_sample *last_sample = NULL;
+    bool healthy;
+    bool rv;
+
+    flb_time_get(&tp);
+    timestamp_seconds = flb_time_to_nanosec(&tp) / 1000000000;
```

Comment:
Maybe use `floor` and `flb_time_to_double` or even add a function in `flb_time.c` to avoid repeating this code pattern.